From 23dcd4339428e4080029ec5ae658e75f01e79a62 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C4=90o=C3=A0n=20Tr=E1=BA=A7n=20C=C3=B4ng=20Danh?=
 <congdanhqx@gmail.com>
Date: Sun, 27 Nov 2022 18:59:24 +0700
Subject: [PATCH] i18n: fix build with Boost 1.81.0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Boost.Locale since 1.81.0 switch to enum classes for facet type [1].
Switch our resource manager accordingly.

1: https://github.com/boostorg/locale/commit/e5ed439ea39a4de915e9fb939cca95cae7d9d8d1

Change-Id: Ief215363ceb79b2019606b20a7cde55c8df6a042
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/143343
Reviewed-by: Heiko Becker <heirecka@exherbo.org>
Tested-by: Caolán McNamara <caolanm@redhat.com>
Reviewed-by: Caolán McNamara <caolanm@redhat.com>
---
 unotools/source/i18n/resmgr.cxx | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/unotools/source/i18n/resmgr.cxx b/unotools/source/i18n/resmgr.cxx
index ddedf5b9e8a2f..3a71163500624 100644
--- a/unotools/source/i18n/resmgr.cxx
+++ b/unotools/source/i18n/resmgr.cxx
@@ -127,8 +127,13 @@ namespace Translate
         if (aFind != aCache.end())
             return aFind->second;
         boost::locale::generator gen;
+#if BOOST_VERSION < 108100
         gen.characters(boost::locale::char_facet);
         gen.categories(boost::locale::message_facet | boost::locale::information_facet);
+#else
+        gen.characters(boost::locale::char_facet_t::char_f);
+        gen.categories(boost::locale::category_t::message | boost::locale::category_t::information);
+#endif
 #if defined(ANDROID)
         OString sPath(OString(lo_get_app_data_dir()) + "/program/resource");
 #else
